// @hashgraph/cryptography/lib/primitive/aes.native.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CipherAlgorithm = void 0;
exports.createCipheriv = createCipheriv;
exports.createDecipheriv = createDecipheriv;
var _cryptoJs = _interopRequireDefault(require("crypto-js"));
var hex = _interopRequireWildcard(require("../encoding/hex.cjs"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const CipherAlgorithm = {
  Aes128Ctr: "AES-128-CTR",
  Aes128Cbc: "AES-128-CBC"
};

/**
 * @param {string} algorithm
 * @param {Uint8Array} key
 * @param {Uint8Array} iv
 * @param {Uint8Array} data
 * @returns {Promise<Uint8Array>}
 */
exports.CipherAlgorithm = CipherAlgorithm;
function createCipheriv(algorithm, key, iv, data) {
  let mode;
  switch (algorithm.toUpperCase()) {
    case CipherAlgorithm.Aes128Cbc:
      mode = _cryptoJs.default.mode.CBC;
      break;
    case CipherAlgorithm.Aes128Ctr:
      mode = _cryptoJs.default.mode.CTR;
      break;
    default:
      throw new Error("(BUG) non-exhaustive switch statement");
  }
  const data_ = _cryptoJs.default.enc.Hex.parse(hex.encode(data));
  const key_ = _cryptoJs.default.enc.Hex.parse(hex.encode(key.slice(0, 16)));
  const cfg_ = {
    iv: _cryptoJs.default.enc.Hex.parse(hex.encode(iv)),
    mode
  };
  return Promise.resolve(hex.decode(_cryptoJs.default.AES.encrypt(data_, key_, cfg_).toString(_cryptoJs.default.format.Hex)));
}

/**
 * @param {string} algorithm
 * @param {Uint8Array} key
 * @param {Uint8Array} iv
 * @param {Uint8Array} data
 * @returns {Promise<Uint8Array>}
 */
function createDecipheriv(algorithm, key, iv, data) {
  let mode;
  switch (algorithm) {
    case CipherAlgorithm.Aes128Cbc:
      mode = _cryptoJs.default.mode.CBC;
      break;
    case CipherAlgorithm.Aes128Ctr:
      mode = _cryptoJs.default.mode.CTR;
      break;
    default:
      throw new Error("(BUG) non-exhaustive switch statement");
  }
  const key_ = _cryptoJs.default.enc.Hex.parse(hex.encode(key.slice(0, 16)));
  const iv_ = _cryptoJs.default.enc.Hex.parse(hex.encode(iv));
  const params = _cryptoJs.default.lib.CipherParams.create({
    ciphertext: _cryptoJs.default.enc.Hex.parse(hex.encode(data)),
    iv: iv_,
    key: key_,
    algorithm: _cryptoJs.default.algo.AES,
    blockSize: 4
  });
  return Promise.resolve(hex.decode(_cryptoJs.default.AES.decrypt(params, key_, {
    iv: iv_,
    mode
  }).toString(_cryptoJs.default.enc.Hex)));
}
