// @hashgraph/sdk/lib/schedule/ScheduleId.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleId = void 0;
const BasicTypes_pb_1 = require("../generated/BasicTypes_pb");
const util_1 = require("../util");
/** Normalized schedule ID returned by various methods in the SDK. */
class ScheduleId {
    constructor(shardOrScheduleId, realm, schedule) {
        if (typeof shardOrScheduleId === "number" &&
            realm != null &&
            schedule != null) {
            this.shard = shardOrScheduleId;
            this.realm = realm;
            this.schedule = schedule;
        }
        else {
            const scheduleId = shardOrScheduleId;
            const id = scheduleId instanceof ScheduleId ?
                scheduleId :
                util_1.normalizeEntityId("schedule", scheduleId);
            this.shard = id.shard;
            this.realm = id.realm;
            this.schedule = id.schedule;
        }
    }
    static fromString(id) {
        return new ScheduleId(id);
    }
    // NOT A STABLE API
    static _fromProto(scheduleId) {
        return new ScheduleId({
            shard: scheduleId.getShardnum(),
            realm: scheduleId.getRealmnum(),
            schedule: scheduleId.getSchedulenum()
        });
    }
    toString() {
        return `${this.shard}.${this.realm}.${this.schedule}`;
    }
    // NOT A STABLE API
    _toProto() {
        const acctId = new BasicTypes_pb_1.ScheduleID();
        acctId.setShardnum(this.shard);
        acctId.setRealmnum(this.realm);
        acctId.setSchedulenum(this.schedule);
        return acctId;
    }
}
exports.ScheduleId = ScheduleId;
