// @hashgraph/cryptography/lib/EcdsaPublicKey.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _Key = _interopRequireDefault(require("./Key.cjs"));
var _BadKeyError = _interopRequireDefault(require("./BadKeyError.cjs"));
var _array = require("./util/array.cjs");
var hex = _interopRequireWildcard(require("./encoding/hex.cjs"));
var ecdsa = _interopRequireWildcard(require("./primitive/ecdsa.cjs"));
var _keccak = require("./primitive/keccak.cjs");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const derPrefix = "302d300706052b8104000a032200";
const derPrefixBytes = hex.decode(derPrefix);

/**
 * An public key on the Hedera™ network.
 */
class EcdsaPublicKey extends _Key.default {
  /**
   * @internal
   * @hideconstructor
   * @param {Uint8Array} keyData
   */
  constructor(keyData) {
    super();

    /**
     * @type {Uint8Array}
     * @private
     * @readonly
     */
    this._keyData = keyData;
  }

  /**
   * @returns {string}
   */
  get _type() {
    return "secp256k1";
  }

  /**
   * @param {Uint8Array} data
   * @returns {EcdsaPublicKey}
   */
  static fromBytes(data) {
    switch (data.length) {
      case 33:
        return EcdsaPublicKey.fromBytesRaw(data);
      case 47:
        return EcdsaPublicKey.fromBytesDer(data);
      default:
        throw new _BadKeyError.default(`invalid public key length: ${data.length} bytes`);
    }
  }

  /**
   * @param {Uint8Array} data
   * @returns {EcdsaPublicKey}
   */
  static fromBytesDer(data) {
    if (data.length != 47 || !(0, _array.arrayStartsWith)(data, derPrefixBytes)) {
      throw new _BadKeyError.default(`invalid public key length: ${data.length} bytes`);
    }
    return new EcdsaPublicKey(data.subarray(derPrefixBytes.length));
  }

  /**
   * @param {Uint8Array} data
   * @returns {EcdsaPublicKey}
   */
  static fromBytesRaw(data) {
    if (data.length != 33) {
      throw new _BadKeyError.default(`invalid public key length: ${data.length} bytes`);
    }
    return new EcdsaPublicKey(data);
  }

  /**
   * Parse a public key from a string of hexadecimal digits.
   *
   * The public key may optionally be prefixed with
   * the DER header.
   *
   * @param {string} text
   * @returns {EcdsaPublicKey}
   */
  static fromString(text) {
    return EcdsaPublicKey.fromBytes(hex.decode(text));
  }

  /**
   * Verify a signature on a message with this public key.
   *
   * @param {Uint8Array} message
   * @param {Uint8Array} signature
   * @returns {boolean}
   */
  verify(message, signature) {
    return ecdsa.verify(this._keyData, message, signature);
  }

  /**
   * @returns {Uint8Array}
   */
  toBytesDer() {
    const bytes = new Uint8Array(derPrefixBytes.length + this._keyData.length);
    bytes.set(derPrefixBytes, 0);
    bytes.set(this._keyData, derPrefixBytes.length);
    return bytes;
  }

  /**
   * @returns {Uint8Array}
   */
  toBytesRaw() {
    return new Uint8Array(this._keyData.subarray());
  }

  /**
   * @returns {string}
   */
  toEthereumAddress() {
    const hash = hex.decode((0, _keccak.keccak256)(`0x${hex.encode(ecdsa.getFullPublicKey(this.toBytesRaw()).subarray(1))}`));
    return hex.encode(hash.subarray(12));
  }

  /**
   * @param {EcdsaPublicKey} other
   * @returns {boolean}
   */
  equals(other) {
    return (0, _array.arrayEqual)(this._keyData, other._keyData);
  }
}
exports.default = EcdsaPublicKey;
