// @hashgraph/sdk/lib/EvmAddress.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _Key = _interopRequireDefault(require("./Key.cjs"));
var hex = _interopRequireWildcard(require("./encoding/hex.cjs"));
var _util = require("./util.cjs");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*-
 * ‌
 * Hedera JavaScript SDK
 * ​
 * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

/**
 * @namespace proto
 * @typedef {import("@hashgraph/proto").proto.IKey} HashgraphProto.proto.IKey
 */
/**
 * @typedef {import("./client/Client.js").default<*, *>} Client
 */
class EvmAddress extends _Key.default {
  /**
   * @internal
   * @param {Uint8Array} bytes
   */
  constructor(bytes) {
    super();
    this._bytes = bytes;
  }

  /**
   * @param {string} text
   * @returns {EvmAddress}
   */
  static fromString(text) {
    return new EvmAddress(hex.decode(text));
  }

  /**
   * @param {Uint8Array} bytes
   * @returns {EvmAddress}
   */
  static fromBytes(bytes) {
    return new EvmAddress(bytes);
  }

  /**
   * @returns {Uint8Array}
   */
  toBytes() {
    return this._bytes;
  }

  /**
   * @returns {string}
   */
  toString() {
    return hex.encode(this._bytes);
  }

  /**
   * @param {EvmAddress} other
   * @returns {boolean}
   */
  equals(other) {
    return (0, _util.arrayEqual)(this._bytes, other._bytes);
  }
}
exports.default = EvmAddress;
