// @hashgraph/sdk/lib/FeeSchedule.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var HashgraphProto = _interopRequireWildcard(require("@hashgraph/proto"));
var _TransactionFeeSchedule = _interopRequireDefault(require("./TransactionFeeSchedule.cjs"));
var _Timestamp = _interopRequireDefault(require("./Timestamp.cjs"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
/*-
 * ‌
 * Hedera JavaScript SDK
 * ​
 * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

class FeeSchedule {
  /**
   * @param {object} [props]
   * @param {TransactionFeeSchedule[]} [props.transactionFeeSchedule]
   * @param {Timestamp} [props.expirationTime]
   */
  constructor(props = {}) {
    /*
     * List of price coefficients for network resources
     *
     * @type {TransactionFeeSchedule}
     */
    this.transactionFeeSchedule = props.transactionFeeSchedule;

    /*
     * FeeSchedule expiry time
     *
     * @type {Timestamp}
     */
    this.expirationTime = props.expirationTime;
  }

  /**
   * @param {Uint8Array} bytes
   * @returns {FeeSchedule}
   */
  static fromBytes(bytes) {
    return FeeSchedule._fromProtobuf(HashgraphProto.proto.FeeSchedule.decode(bytes));
  }

  /**
   * @internal
   * @param {HashgraphProto.proto.IFeeSchedule} feeSchedule
   * @returns {FeeSchedule}
   */
  static _fromProtobuf(feeSchedule) {
    return new FeeSchedule({
      transactionFeeSchedule: feeSchedule.transactionFeeSchedule != null ? feeSchedule.transactionFeeSchedule.map(schedule => _TransactionFeeSchedule.default._fromProtobuf(schedule)) : undefined,
      expirationTime: feeSchedule.expiryTime != null ? _Timestamp.default._fromProtobuf(feeSchedule.expiryTime) : undefined
    });
  }

  /**
   * @internal
   * @returns {HashgraphProto.proto.IFeeSchedule}
   */
  _toProtobuf() {
    return {
      transactionFeeSchedule: this.transactionFeeSchedule != null ? this.transactionFeeSchedule.map(transaction => transaction._toProtobuf()) : undefined,
      expiryTime: this.expirationTime != null ? this.expirationTime._toProtobuf() : undefined
    };
  }

  /**
   * @returns {Uint8Array}
   */
  toBytes() {
    return HashgraphProto.proto.FeeSchedule.encode(this._toProtobuf()).finish();
  }
}
exports.default = FeeSchedule;
