// @hashgraph/sdk/lib/LocalProvider.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _NodeClient = _interopRequireDefault(require("./client/NodeClient.cjs"));
var _AccountBalanceQuery = _interopRequireDefault(require("./account/AccountBalanceQuery.cjs"));
var _AccountInfoQuery = _interopRequireDefault(require("./account/AccountInfoQuery.cjs"));
var _AccountRecordsQuery = _interopRequireDefault(require("./account/AccountRecordsQuery.cjs"));
var _TransactionReceiptQuery = _interopRequireDefault(require("./transaction/TransactionReceiptQuery.cjs"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*-
 * ‌
 * Hedera JavaScript SDK
 * ​
 * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

/**
 * @typedef {import("./Provider.js").Provider} Provider
 * @typedef {import("./LedgerId.js").default} LedgerId
 * @typedef {import("./Key.js").default} Key
 * @typedef {import("./transaction/Transaction.js").default} Transaction
 * @typedef {import("./transaction/TransactionId.js").default} TransactionId
 * @typedef {import("./transaction/TransactionResponse.js").default} TransactionResponse
 * @typedef {import("./transaction/TransactionReceipt.js").default} TransactionReceipt
 * @typedef {import("./transaction/TransactionRecord.js").default} TransactionRecord
 * @typedef {import("./account/AccountId.js").default} AccountId
 * @typedef {import("./account/AccountBalance.js").default} AccountBalance
 * @typedef {import("./account/AccountInfo.js").default} AccountInfo
 */

/**
 * @template {any} O
 * @typedef {import("./query/Query.js").default<O>} Query<O>
 */

/**
 * @template RequestT
 * @template ResponseT
 * @template OutputT
 * @typedef {import("./Executable.js").default<RequestT, ResponseT, OutputT>} Executable<RequestT, ResponseT, OutputT>
 */
/**
 * @implements {Provider}
 */
class LocalProvider {
  /**
   * @param {object} props
   * @param {Client} [props.client]
   */
  constructor(props = {}) {
    if (props != null && props.client != null) {
      this._client = props.client;
      return;
    }
    if (process.env.HEDERA_NETWORK == null) {
      throw new Error("LocalProvider requires the `HEDERA_NETWORK` environment variable to be set");
    }
    this._client = _NodeClient.default.forName(process.env.HEDERA_NETWORK);
  }

  /**
   * @param {Client} client
   * @returns {LocalProvider}
   */
  static fromClient(client) {
    return new LocalProvider({
      client
    });
  }

  /**
   * @returns {LedgerId?}
   */
  getLedgerId() {
    return this._client.ledgerId;
  }

  /**
   * @returns {{[key: string]: (string | AccountId)}}
   */
  getNetwork() {
    return this._client.network;
  }

  /**
   * @returns {string[]}
   */
  getMirrorNetwork() {
    return this._client.mirrorNetwork;
  }

  /**
   * @param {AccountId | string} accountId
   * @returns {Promise<AccountBalance>}
   */
  getAccountBalance(accountId) {
    return new _AccountBalanceQuery.default().setAccountId(accountId).execute(this._client);
  }

  /**
   * @param {AccountId | string} accountId
   * @returns {Promise<AccountInfo>}
   */
  getAccountInfo(accountId) {
    return new _AccountInfoQuery.default().setAccountId(accountId).execute(this._client);
  }

  /**
   * @param {AccountId | string} accountId
   * @returns {Promise<TransactionRecord[]>}
   */
  getAccountRecords(accountId) {
    return new _AccountRecordsQuery.default().setAccountId(accountId).execute(this._client);
  }

  /**
   * @param {TransactionId | string} transactionId
   * @returns {Promise<TransactionReceipt>}
   */
  getTransactionReceipt(transactionId) {
    return new _TransactionReceiptQuery.default().setTransactionId(transactionId).execute(this._client);
  }

  /**
   * @param {TransactionResponse} response
   * @returns {Promise<TransactionReceipt>}
   */
  waitForReceipt(response) {
    return new _TransactionReceiptQuery.default().setNodeAccountIds([response.nodeId]).setTransactionId(response.transactionId).execute(this._client);
  }

  /**
   * @template RequestT
   * @template ResponseT
   * @template OutputT
   * @param {Executable<RequestT, ResponseT, OutputT>} request
   * @returns {Promise<OutputT>}
   */
  call(request) {
    return request.execute(this._client);
  }
}
exports.default = LocalProvider;
