// @hashgraph/sdk/lib/Node.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _ManagedNode = _interopRequireDefault(require("./ManagedNode.cjs"));
var _NodeCerts = require("./NodeCerts.cjs");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/*-
 * ‌
 * Hedera JavaScript SDK
 * ​
 * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

/**
 * @typedef {import("./account/AccountId.js").default} AccountId
 * @typedef {import("./address_book/NodeAddress.js").default} NodeAddress
 * @typedef {import("./channel/Channel.js").default} Channel
 * @typedef {import("./ManagedNodeAddress.js").default} ManagedNodeAddress
 * @typedef {import("./LedgerId.js").default} LedgerId
 */

/**
 * @typedef {object} NewNode
 * @property {AccountId} accountId
 * @property {string} address
 * @property {(address: string, cert?: string) => Channel} channelInitFunction
 */

/**
 * @typedef {object} CloneNode
 * @property {Node} node
 * @property {ManagedNodeAddress} address
 */
/**
 * @augments {ManagedNode<Channel>}
 */
class Node extends _ManagedNode.default {
  /**
   * @param {object} props
   * @param {NewNode=} [props.newNode]
   * @param {CloneNode=} [props.cloneNode]
   */
  constructor(props = {}) {
    super(props);
    if (props.newNode != null) {
      /** @type {AccountId} */
      this._accountId = props.newNode.accountId;

      /** @type {NodeAddress | null} */
      this._nodeAddress = null;
    } else if (props.cloneNode != null) {
      /** @type {AccountId} */
      this._accountId = props.cloneNode.node._accountId;

      /** @type {NodeAddress | null} */
      this._nodeAddress = props.cloneNode.node._nodeAddress;
    } else {
      throw new Error(`failed to create node: ${JSON.stringify(props)}`);
    }
  }

  /**
   * @returns {string}
   */
  getKey() {
    return this._accountId.toString();
  }

  /**
   * @returns {ManagedNode<Channel>}
   */
  toInsecure() {
    return (/** @type {this} */
      new Node({
        cloneNode: {
          node: this,
          address: this._address.toInsecure()
        }
      })
    );
  }

  /**
   * @returns {ManagedNode<Channel>}
   */
  toSecure() {
    return (/** @type {this} */
      new Node({
        cloneNode: {
          node: this,
          address: this._address.toSecure()
        }
      })
    );
  }

  /**
   * @param {LedgerId|string} ledgerId
   * @returns {this}
   */
  setCert(ledgerId) {
    switch (ledgerId.toString()) {
      case "previewnet":
        this._cert = _NodeCerts.PREVIEWNET_CERTS[this._accountId.toString()];
        break;
      case "testnet":
        this._cert = _NodeCerts.TESTNET_CERTS[this._accountId.toString()];
        break;
      case "mainnet":
        this._cert = _NodeCerts.MAINNET_CERTS[this._accountId.toString()];
        break;
    }
    return this;
  }

  /**
   * @returns {AccountId}
   */
  get accountId() {
    return this._accountId;
  }

  /**
   * @returns {NodeAddress | null}
   */
  get nodeAddress() {
    return this._nodeAddress;
  }

  /**
   * @param {NodeAddress} nodeAddress
   * @returns {this}
   */
  setNodeAddress(nodeAddress) {
    this._nodeAddress = nodeAddress;
    return this;
  }
}
exports.default = Node;
