// @hashgraph/sdk/lib/generated/CryptoTransfer_pb.js
// source: CryptoTransfer.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
goog.exportSymbol('proto.proto.CryptoTransferTransactionBody', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.CryptoTransferTransactionBody = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.CryptoTransferTransactionBody.repeatedFields_, null);
};
goog.inherits(proto.proto.CryptoTransferTransactionBody, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.CryptoTransferTransactionBody.displayName = 'proto.proto.CryptoTransferTransactionBody';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.CryptoTransferTransactionBody.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.CryptoTransferTransactionBody.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.CryptoTransferTransactionBody.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.CryptoTransferTransactionBody} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoTransferTransactionBody.toObject = function(includeInstance, msg) {
  var f, obj = {
    transfers: (f = msg.getTransfers()) && BasicTypes_pb.TransferList.toObject(includeInstance, f),
    tokentransfersList: jspb.Message.toObjectList(msg.getTokentransfersList(),
    BasicTypes_pb.TokenTransferList.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.CryptoTransferTransactionBody}
 */
proto.proto.CryptoTransferTransactionBody.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.CryptoTransferTransactionBody;
  return proto.proto.CryptoTransferTransactionBody.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.CryptoTransferTransactionBody} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.CryptoTransferTransactionBody}
 */
proto.proto.CryptoTransferTransactionBody.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new BasicTypes_pb.TransferList;
      reader.readMessage(value,BasicTypes_pb.TransferList.deserializeBinaryFromReader);
      msg.setTransfers(value);
      break;
    case 2:
      var value = new BasicTypes_pb.TokenTransferList;
      reader.readMessage(value,BasicTypes_pb.TokenTransferList.deserializeBinaryFromReader);
      msg.addTokentransfers(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.CryptoTransferTransactionBody.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.CryptoTransferTransactionBody.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.CryptoTransferTransactionBody} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoTransferTransactionBody.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTransfers();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      BasicTypes_pb.TransferList.serializeBinaryToWriter
    );
  }
  f = message.getTokentransfersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      BasicTypes_pb.TokenTransferList.serializeBinaryToWriter
    );
  }
};


/**
 * optional TransferList transfers = 1;
 * @return {?proto.proto.TransferList}
 */
proto.proto.CryptoTransferTransactionBody.prototype.getTransfers = function() {
  return /** @type{?proto.proto.TransferList} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.TransferList, 1));
};


/**
 * @param {?proto.proto.TransferList|undefined} value
 * @return {!proto.proto.CryptoTransferTransactionBody} returns this
*/
proto.proto.CryptoTransferTransactionBody.prototype.setTransfers = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoTransferTransactionBody} returns this
 */
proto.proto.CryptoTransferTransactionBody.prototype.clearTransfers = function() {
  return this.setTransfers(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoTransferTransactionBody.prototype.hasTransfers = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated TokenTransferList tokenTransfers = 2;
 * @return {!Array<!proto.proto.TokenTransferList>}
 */
proto.proto.CryptoTransferTransactionBody.prototype.getTokentransfersList = function() {
  return /** @type{!Array<!proto.proto.TokenTransferList>} */ (
    jspb.Message.getRepeatedWrapperField(this, BasicTypes_pb.TokenTransferList, 2));
};


/**
 * @param {!Array<!proto.proto.TokenTransferList>} value
 * @return {!proto.proto.CryptoTransferTransactionBody} returns this
*/
proto.proto.CryptoTransferTransactionBody.prototype.setTokentransfersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.proto.TokenTransferList=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.TokenTransferList}
 */
proto.proto.CryptoTransferTransactionBody.prototype.addTokentransfers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.proto.TokenTransferList, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.CryptoTransferTransactionBody} returns this
 */
proto.proto.CryptoTransferTransactionBody.prototype.clearTokentransfersList = function() {
  return this.setTokentransfersList([]);
};


goog.object.extend(exports, proto.proto);
