// @hashgraph/sdk/lib/generated/CryptoGetInfo_pb.js
// source: CryptoGetInfo.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var Timestamp_pb = require('./Timestamp_pb.js');
goog.object.extend(proto, Timestamp_pb);
var Duration_pb = require('./Duration_pb.js');
goog.object.extend(proto, Duration_pb);
var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
var QueryHeader_pb = require('./QueryHeader_pb.js');
goog.object.extend(proto, QueryHeader_pb);
var ResponseHeader_pb = require('./ResponseHeader_pb.js');
goog.object.extend(proto, ResponseHeader_pb);
var CryptoAddLiveHash_pb = require('./CryptoAddLiveHash_pb.js');
goog.object.extend(proto, CryptoAddLiveHash_pb);
goog.exportSymbol('proto.proto.CryptoGetInfoQuery', null, global);
goog.exportSymbol('proto.proto.CryptoGetInfoResponse', null, global);
goog.exportSymbol('proto.proto.CryptoGetInfoResponse.AccountInfo', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.CryptoGetInfoQuery = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.CryptoGetInfoQuery, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.CryptoGetInfoQuery.displayName = 'proto.proto.CryptoGetInfoQuery';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.CryptoGetInfoResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.CryptoGetInfoResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.CryptoGetInfoResponse.displayName = 'proto.proto.CryptoGetInfoResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.CryptoGetInfoResponse.AccountInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.CryptoGetInfoResponse.AccountInfo.repeatedFields_, null);
};
goog.inherits(proto.proto.CryptoGetInfoResponse.AccountInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.CryptoGetInfoResponse.AccountInfo.displayName = 'proto.proto.CryptoGetInfoResponse.AccountInfo';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.CryptoGetInfoQuery.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.CryptoGetInfoQuery.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.CryptoGetInfoQuery} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoQuery.toObject = function(includeInstance, msg) {
  var f, obj = {
    header: (f = msg.getHeader()) && QueryHeader_pb.QueryHeader.toObject(includeInstance, f),
    accountid: (f = msg.getAccountid()) && BasicTypes_pb.AccountID.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.CryptoGetInfoQuery}
 */
proto.proto.CryptoGetInfoQuery.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.CryptoGetInfoQuery;
  return proto.proto.CryptoGetInfoQuery.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.CryptoGetInfoQuery} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.CryptoGetInfoQuery}
 */
proto.proto.CryptoGetInfoQuery.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new QueryHeader_pb.QueryHeader;
      reader.readMessage(value,QueryHeader_pb.QueryHeader.deserializeBinaryFromReader);
      msg.setHeader(value);
      break;
    case 2:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setAccountid(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.CryptoGetInfoQuery.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.CryptoGetInfoQuery.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.CryptoGetInfoQuery} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoQuery.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getHeader();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      QueryHeader_pb.QueryHeader.serializeBinaryToWriter
    );
  }
  f = message.getAccountid();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
};


/**
 * optional QueryHeader header = 1;
 * @return {?proto.proto.QueryHeader}
 */
proto.proto.CryptoGetInfoQuery.prototype.getHeader = function() {
  return /** @type{?proto.proto.QueryHeader} */ (
    jspb.Message.getWrapperField(this, QueryHeader_pb.QueryHeader, 1));
};


/**
 * @param {?proto.proto.QueryHeader|undefined} value
 * @return {!proto.proto.CryptoGetInfoQuery} returns this
*/
proto.proto.CryptoGetInfoQuery.prototype.setHeader = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoQuery} returns this
 */
proto.proto.CryptoGetInfoQuery.prototype.clearHeader = function() {
  return this.setHeader(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoQuery.prototype.hasHeader = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional AccountID accountID = 2;
 * @return {?proto.proto.AccountID}
 */
proto.proto.CryptoGetInfoQuery.prototype.getAccountid = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 2));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.CryptoGetInfoQuery} returns this
*/
proto.proto.CryptoGetInfoQuery.prototype.setAccountid = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoQuery} returns this
 */
proto.proto.CryptoGetInfoQuery.prototype.clearAccountid = function() {
  return this.setAccountid(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoQuery.prototype.hasAccountid = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.CryptoGetInfoResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.CryptoGetInfoResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.CryptoGetInfoResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    header: (f = msg.getHeader()) && ResponseHeader_pb.ResponseHeader.toObject(includeInstance, f),
    accountinfo: (f = msg.getAccountinfo()) && proto.proto.CryptoGetInfoResponse.AccountInfo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.CryptoGetInfoResponse}
 */
proto.proto.CryptoGetInfoResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.CryptoGetInfoResponse;
  return proto.proto.CryptoGetInfoResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.CryptoGetInfoResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.CryptoGetInfoResponse}
 */
proto.proto.CryptoGetInfoResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new ResponseHeader_pb.ResponseHeader;
      reader.readMessage(value,ResponseHeader_pb.ResponseHeader.deserializeBinaryFromReader);
      msg.setHeader(value);
      break;
    case 2:
      var value = new proto.proto.CryptoGetInfoResponse.AccountInfo;
      reader.readMessage(value,proto.proto.CryptoGetInfoResponse.AccountInfo.deserializeBinaryFromReader);
      msg.setAccountinfo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.CryptoGetInfoResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.CryptoGetInfoResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.CryptoGetInfoResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getHeader();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      ResponseHeader_pb.ResponseHeader.serializeBinaryToWriter
    );
  }
  f = message.getAccountinfo();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.proto.CryptoGetInfoResponse.AccountInfo.serializeBinaryToWriter
    );
  }
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.repeatedFields_ = [14,15];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.CryptoGetInfoResponse.AccountInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.CryptoGetInfoResponse.AccountInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    accountid: (f = msg.getAccountid()) && BasicTypes_pb.AccountID.toObject(includeInstance, f),
    contractaccountid: jspb.Message.getFieldWithDefault(msg, 2, ""),
    deleted: jspb.Message.getBooleanFieldWithDefault(msg, 3, false),
    proxyaccountid: (f = msg.getProxyaccountid()) && BasicTypes_pb.AccountID.toObject(includeInstance, f),
    proxyreceived: jspb.Message.getFieldWithDefault(msg, 6, "0"),
    key: (f = msg.getKey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    balance: jspb.Message.getFieldWithDefault(msg, 8, "0"),
    generatesendrecordthreshold: jspb.Message.getFieldWithDefault(msg, 9, "0"),
    generatereceiverecordthreshold: jspb.Message.getFieldWithDefault(msg, 10, "0"),
    receiversigrequired: jspb.Message.getBooleanFieldWithDefault(msg, 11, false),
    expirationtime: (f = msg.getExpirationtime()) && Timestamp_pb.Timestamp.toObject(includeInstance, f),
    autorenewperiod: (f = msg.getAutorenewperiod()) && Duration_pb.Duration.toObject(includeInstance, f),
    livehashesList: jspb.Message.toObjectList(msg.getLivehashesList(),
    CryptoAddLiveHash_pb.LiveHash.toObject, includeInstance),
    tokenrelationshipsList: jspb.Message.toObjectList(msg.getTokenrelationshipsList(),
    BasicTypes_pb.TokenRelationship.toObject, includeInstance),
    memo: jspb.Message.getFieldWithDefault(msg, 16, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.CryptoGetInfoResponse.AccountInfo;
  return proto.proto.CryptoGetInfoResponse.AccountInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.CryptoGetInfoResponse.AccountInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setAccountid(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setContractaccountid(value);
      break;
    case 3:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setDeleted(value);
      break;
    case 4:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setProxyaccountid(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readInt64String());
      msg.setProxyreceived(value);
      break;
    case 7:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setKey(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readUint64String());
      msg.setBalance(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readUint64String());
      msg.setGeneratesendrecordthreshold(value);
      break;
    case 10:
      var value = /** @type {string} */ (reader.readUint64String());
      msg.setGeneratereceiverecordthreshold(value);
      break;
    case 11:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setReceiversigrequired(value);
      break;
    case 12:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setExpirationtime(value);
      break;
    case 13:
      var value = new Duration_pb.Duration;
      reader.readMessage(value,Duration_pb.Duration.deserializeBinaryFromReader);
      msg.setAutorenewperiod(value);
      break;
    case 14:
      var value = new CryptoAddLiveHash_pb.LiveHash;
      reader.readMessage(value,CryptoAddLiveHash_pb.LiveHash.deserializeBinaryFromReader);
      msg.addLivehashes(value);
      break;
    case 15:
      var value = new BasicTypes_pb.TokenRelationship;
      reader.readMessage(value,BasicTypes_pb.TokenRelationship.deserializeBinaryFromReader);
      msg.addTokenrelationships(value);
      break;
    case 16:
      var value = /** @type {string} */ (reader.readString());
      msg.setMemo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.CryptoGetInfoResponse.AccountInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.CryptoGetInfoResponse.AccountInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAccountid();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
  f = message.getContractaccountid();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getDeleted();
  if (f) {
    writer.writeBool(
      3,
      f
    );
  }
  f = message.getProxyaccountid();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
  f = message.getProxyreceived();
  if (parseInt(f, 10) !== 0) {
    writer.writeInt64String(
      6,
      f
    );
  }
  f = message.getKey();
  if (f != null) {
    writer.writeMessage(
      7,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getBalance();
  if (parseInt(f, 10) !== 0) {
    writer.writeUint64String(
      8,
      f
    );
  }
  f = message.getGeneratesendrecordthreshold();
  if (parseInt(f, 10) !== 0) {
    writer.writeUint64String(
      9,
      f
    );
  }
  f = message.getGeneratereceiverecordthreshold();
  if (parseInt(f, 10) !== 0) {
    writer.writeUint64String(
      10,
      f
    );
  }
  f = message.getReceiversigrequired();
  if (f) {
    writer.writeBool(
      11,
      f
    );
  }
  f = message.getExpirationtime();
  if (f != null) {
    writer.writeMessage(
      12,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getAutorenewperiod();
  if (f != null) {
    writer.writeMessage(
      13,
      f,
      Duration_pb.Duration.serializeBinaryToWriter
    );
  }
  f = message.getLivehashesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      14,
      f,
      CryptoAddLiveHash_pb.LiveHash.serializeBinaryToWriter
    );
  }
  f = message.getTokenrelationshipsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      15,
      f,
      BasicTypes_pb.TokenRelationship.serializeBinaryToWriter
    );
  }
  f = message.getMemo();
  if (f.length > 0) {
    writer.writeString(
      16,
      f
    );
  }
};


/**
 * optional AccountID accountID = 1;
 * @return {?proto.proto.AccountID}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getAccountid = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 1));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setAccountid = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearAccountid = function() {
  return this.setAccountid(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.hasAccountid = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string contractAccountID = 2;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getContractaccountid = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setContractaccountid = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional bool deleted = 3;
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getDeleted = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 3, false));
};


/**
 * @param {boolean} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setDeleted = function(value) {
  return jspb.Message.setProto3BooleanField(this, 3, value);
};


/**
 * optional AccountID proxyAccountID = 4;
 * @return {?proto.proto.AccountID}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getProxyaccountid = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 4));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setProxyaccountid = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearProxyaccountid = function() {
  return this.setProxyaccountid(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.hasProxyaccountid = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional int64 proxyReceived = 6;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getProxyreceived = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, "0"));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setProxyreceived = function(value) {
  return jspb.Message.setProto3StringIntField(this, 6, value);
};


/**
 * optional Key key = 7;
 * @return {?proto.proto.Key}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getKey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 7));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setKey = function(value) {
  return jspb.Message.setWrapperField(this, 7, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearKey = function() {
  return this.setKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.hasKey = function() {
  return jspb.Message.getField(this, 7) != null;
};


/**
 * optional uint64 balance = 8;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getBalance = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, "0"));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setBalance = function(value) {
  return jspb.Message.setProto3StringIntField(this, 8, value);
};


/**
 * optional uint64 generateSendRecordThreshold = 9;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getGeneratesendrecordthreshold = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, "0"));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setGeneratesendrecordthreshold = function(value) {
  return jspb.Message.setProto3StringIntField(this, 9, value);
};


/**
 * optional uint64 generateReceiveRecordThreshold = 10;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getGeneratereceiverecordthreshold = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 10, "0"));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setGeneratereceiverecordthreshold = function(value) {
  return jspb.Message.setProto3StringIntField(this, 10, value);
};


/**
 * optional bool receiverSigRequired = 11;
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getReceiversigrequired = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 11, false));
};


/**
 * @param {boolean} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setReceiversigrequired = function(value) {
  return jspb.Message.setProto3BooleanField(this, 11, value);
};


/**
 * optional Timestamp expirationTime = 12;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getExpirationtime = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 12));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setExpirationtime = function(value) {
  return jspb.Message.setWrapperField(this, 12, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearExpirationtime = function() {
  return this.setExpirationtime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.hasExpirationtime = function() {
  return jspb.Message.getField(this, 12) != null;
};


/**
 * optional Duration autoRenewPeriod = 13;
 * @return {?proto.proto.Duration}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getAutorenewperiod = function() {
  return /** @type{?proto.proto.Duration} */ (
    jspb.Message.getWrapperField(this, Duration_pb.Duration, 13));
};


/**
 * @param {?proto.proto.Duration|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setAutorenewperiod = function(value) {
  return jspb.Message.setWrapperField(this, 13, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearAutorenewperiod = function() {
  return this.setAutorenewperiod(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.hasAutorenewperiod = function() {
  return jspb.Message.getField(this, 13) != null;
};


/**
 * repeated LiveHash liveHashes = 14;
 * @return {!Array<!proto.proto.LiveHash>}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getLivehashesList = function() {
  return /** @type{!Array<!proto.proto.LiveHash>} */ (
    jspb.Message.getRepeatedWrapperField(this, CryptoAddLiveHash_pb.LiveHash, 14));
};


/**
 * @param {!Array<!proto.proto.LiveHash>} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setLivehashesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 14, value);
};


/**
 * @param {!proto.proto.LiveHash=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.LiveHash}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.addLivehashes = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 14, opt_value, proto.proto.LiveHash, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearLivehashesList = function() {
  return this.setLivehashesList([]);
};


/**
 * repeated TokenRelationship tokenRelationships = 15;
 * @return {!Array<!proto.proto.TokenRelationship>}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getTokenrelationshipsList = function() {
  return /** @type{!Array<!proto.proto.TokenRelationship>} */ (
    jspb.Message.getRepeatedWrapperField(this, BasicTypes_pb.TokenRelationship, 15));
};


/**
 * @param {!Array<!proto.proto.TokenRelationship>} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
*/
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setTokenrelationshipsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 15, value);
};


/**
 * @param {!proto.proto.TokenRelationship=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.TokenRelationship}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.addTokenrelationships = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 15, opt_value, proto.proto.TokenRelationship, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.clearTokenrelationshipsList = function() {
  return this.setTokenrelationshipsList([]);
};


/**
 * optional string memo = 16;
 * @return {string}
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.getMemo = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 16, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.CryptoGetInfoResponse.AccountInfo} returns this
 */
proto.proto.CryptoGetInfoResponse.AccountInfo.prototype.setMemo = function(value) {
  return jspb.Message.setProto3StringField(this, 16, value);
};


/**
 * optional ResponseHeader header = 1;
 * @return {?proto.proto.ResponseHeader}
 */
proto.proto.CryptoGetInfoResponse.prototype.getHeader = function() {
  return /** @type{?proto.proto.ResponseHeader} */ (
    jspb.Message.getWrapperField(this, ResponseHeader_pb.ResponseHeader, 1));
};


/**
 * @param {?proto.proto.ResponseHeader|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse} returns this
*/
proto.proto.CryptoGetInfoResponse.prototype.setHeader = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse} returns this
 */
proto.proto.CryptoGetInfoResponse.prototype.clearHeader = function() {
  return this.setHeader(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.prototype.hasHeader = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional AccountInfo accountInfo = 2;
 * @return {?proto.proto.CryptoGetInfoResponse.AccountInfo}
 */
proto.proto.CryptoGetInfoResponse.prototype.getAccountinfo = function() {
  return /** @type{?proto.proto.CryptoGetInfoResponse.AccountInfo} */ (
    jspb.Message.getWrapperField(this, proto.proto.CryptoGetInfoResponse.AccountInfo, 2));
};


/**
 * @param {?proto.proto.CryptoGetInfoResponse.AccountInfo|undefined} value
 * @return {!proto.proto.CryptoGetInfoResponse} returns this
*/
proto.proto.CryptoGetInfoResponse.prototype.setAccountinfo = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.CryptoGetInfoResponse} returns this
 */
proto.proto.CryptoGetInfoResponse.prototype.clearAccountinfo = function() {
  return this.setAccountinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.CryptoGetInfoResponse.prototype.hasAccountinfo = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.proto);
