// @hashgraph/sdk/lib/generated/Query_pb.js
// source: Query.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var GetByKey_pb = require('./GetByKey_pb.js');
goog.object.extend(proto, GetByKey_pb);
var GetBySolidityID_pb = require('./GetBySolidityID_pb.js');
goog.object.extend(proto, GetBySolidityID_pb);
var ContractCallLocal_pb = require('./ContractCallLocal_pb.js');
goog.object.extend(proto, ContractCallLocal_pb);
var ContractGetInfo_pb = require('./ContractGetInfo_pb.js');
goog.object.extend(proto, ContractGetInfo_pb);
var ContractGetBytecode_pb = require('./ContractGetBytecode_pb.js');
goog.object.extend(proto, ContractGetBytecode_pb);
var ContractGetRecords_pb = require('./ContractGetRecords_pb.js');
goog.object.extend(proto, ContractGetRecords_pb);
var CryptoGetAccountBalance_pb = require('./CryptoGetAccountBalance_pb.js');
goog.object.extend(proto, CryptoGetAccountBalance_pb);
var CryptoGetAccountRecords_pb = require('./CryptoGetAccountRecords_pb.js');
goog.object.extend(proto, CryptoGetAccountRecords_pb);
var CryptoGetInfo_pb = require('./CryptoGetInfo_pb.js');
goog.object.extend(proto, CryptoGetInfo_pb);
var CryptoGetLiveHash_pb = require('./CryptoGetLiveHash_pb.js');
goog.object.extend(proto, CryptoGetLiveHash_pb);
var CryptoGetStakers_pb = require('./CryptoGetStakers_pb.js');
goog.object.extend(proto, CryptoGetStakers_pb);
var FileGetContents_pb = require('./FileGetContents_pb.js');
goog.object.extend(proto, FileGetContents_pb);
var FileGetInfo_pb = require('./FileGetInfo_pb.js');
goog.object.extend(proto, FileGetInfo_pb);
var TransactionGetReceipt_pb = require('./TransactionGetReceipt_pb.js');
goog.object.extend(proto, TransactionGetReceipt_pb);
var TransactionGetRecord_pb = require('./TransactionGetRecord_pb.js');
goog.object.extend(proto, TransactionGetRecord_pb);
var TransactionGetFastRecord_pb = require('./TransactionGetFastRecord_pb.js');
goog.object.extend(proto, TransactionGetFastRecord_pb);
var ConsensusGetTopicInfo_pb = require('./ConsensusGetTopicInfo_pb.js');
goog.object.extend(proto, ConsensusGetTopicInfo_pb);
var NetworkGetVersionInfo_pb = require('./NetworkGetVersionInfo_pb.js');
goog.object.extend(proto, NetworkGetVersionInfo_pb);
var TokenGetInfo_pb = require('./TokenGetInfo_pb.js');
goog.object.extend(proto, TokenGetInfo_pb);
var ScheduleGetInfo_pb = require('./ScheduleGetInfo_pb.js');
goog.object.extend(proto, ScheduleGetInfo_pb);
goog.exportSymbol('proto.proto.Query', null, global);
goog.exportSymbol('proto.proto.Query.QueryCase', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.Query = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.proto.Query.oneofGroups_);
};
goog.inherits(proto.proto.Query, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.Query.displayName = 'proto.proto.Query';
}

/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.proto.Query.oneofGroups_ = [[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,50,51,52,53]];

/**
 * @enum {number}
 */
proto.proto.Query.QueryCase = {
  QUERY_NOT_SET: 0,
  GETBYKEY: 1,
  GETBYSOLIDITYID: 2,
  CONTRACTCALLLOCAL: 3,
  CONTRACTGETINFO: 4,
  CONTRACTGETBYTECODE: 5,
  CONTRACTGETRECORDS: 6,
  CRYPTOGETACCOUNTBALANCE: 7,
  CRYPTOGETACCOUNTRECORDS: 8,
  CRYPTOGETINFO: 9,
  CRYPTOGETLIVEHASH: 10,
  CRYPTOGETPROXYSTAKERS: 11,
  FILEGETCONTENTS: 12,
  FILEGETINFO: 13,
  TRANSACTIONGETRECEIPT: 14,
  TRANSACTIONGETRECORD: 15,
  TRANSACTIONGETFASTRECORD: 16,
  CONSENSUSGETTOPICINFO: 50,
  NETWORKGETVERSIONINFO: 51,
  TOKENGETINFO: 52,
  SCHEDULEGETINFO: 53
};

/**
 * @return {proto.proto.Query.QueryCase}
 */
proto.proto.Query.prototype.getQueryCase = function() {
  return /** @type {proto.proto.Query.QueryCase} */(jspb.Message.computeOneofCase(this, proto.proto.Query.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.Query.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.Query.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.Query} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.Query.toObject = function(includeInstance, msg) {
  var f, obj = {
    getbykey: (f = msg.getGetbykey()) && GetByKey_pb.GetByKeyQuery.toObject(includeInstance, f),
    getbysolidityid: (f = msg.getGetbysolidityid()) && GetBySolidityID_pb.GetBySolidityIDQuery.toObject(includeInstance, f),
    contractcalllocal: (f = msg.getContractcalllocal()) && ContractCallLocal_pb.ContractCallLocalQuery.toObject(includeInstance, f),
    contractgetinfo: (f = msg.getContractgetinfo()) && ContractGetInfo_pb.ContractGetInfoQuery.toObject(includeInstance, f),
    contractgetbytecode: (f = msg.getContractgetbytecode()) && ContractGetBytecode_pb.ContractGetBytecodeQuery.toObject(includeInstance, f),
    contractgetrecords: (f = msg.getContractgetrecords()) && ContractGetRecords_pb.ContractGetRecordsQuery.toObject(includeInstance, f),
    cryptogetaccountbalance: (f = msg.getCryptogetaccountbalance()) && CryptoGetAccountBalance_pb.CryptoGetAccountBalanceQuery.toObject(includeInstance, f),
    cryptogetaccountrecords: (f = msg.getCryptogetaccountrecords()) && CryptoGetAccountRecords_pb.CryptoGetAccountRecordsQuery.toObject(includeInstance, f),
    cryptogetinfo: (f = msg.getCryptogetinfo()) && CryptoGetInfo_pb.CryptoGetInfoQuery.toObject(includeInstance, f),
    cryptogetlivehash: (f = msg.getCryptogetlivehash()) && CryptoGetLiveHash_pb.CryptoGetLiveHashQuery.toObject(includeInstance, f),
    cryptogetproxystakers: (f = msg.getCryptogetproxystakers()) && CryptoGetStakers_pb.CryptoGetStakersQuery.toObject(includeInstance, f),
    filegetcontents: (f = msg.getFilegetcontents()) && FileGetContents_pb.FileGetContentsQuery.toObject(includeInstance, f),
    filegetinfo: (f = msg.getFilegetinfo()) && FileGetInfo_pb.FileGetInfoQuery.toObject(includeInstance, f),
    transactiongetreceipt: (f = msg.getTransactiongetreceipt()) && TransactionGetReceipt_pb.TransactionGetReceiptQuery.toObject(includeInstance, f),
    transactiongetrecord: (f = msg.getTransactiongetrecord()) && TransactionGetRecord_pb.TransactionGetRecordQuery.toObject(includeInstance, f),
    transactiongetfastrecord: (f = msg.getTransactiongetfastrecord()) && TransactionGetFastRecord_pb.TransactionGetFastRecordQuery.toObject(includeInstance, f),
    consensusgettopicinfo: (f = msg.getConsensusgettopicinfo()) && ConsensusGetTopicInfo_pb.ConsensusGetTopicInfoQuery.toObject(includeInstance, f),
    networkgetversioninfo: (f = msg.getNetworkgetversioninfo()) && NetworkGetVersionInfo_pb.NetworkGetVersionInfoQuery.toObject(includeInstance, f),
    tokengetinfo: (f = msg.getTokengetinfo()) && TokenGetInfo_pb.TokenGetInfoQuery.toObject(includeInstance, f),
    schedulegetinfo: (f = msg.getSchedulegetinfo()) && ScheduleGetInfo_pb.ScheduleGetInfoQuery.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.Query}
 */
proto.proto.Query.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.Query;
  return proto.proto.Query.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.Query} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.Query}
 */
proto.proto.Query.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new GetByKey_pb.GetByKeyQuery;
      reader.readMessage(value,GetByKey_pb.GetByKeyQuery.deserializeBinaryFromReader);
      msg.setGetbykey(value);
      break;
    case 2:
      var value = new GetBySolidityID_pb.GetBySolidityIDQuery;
      reader.readMessage(value,GetBySolidityID_pb.GetBySolidityIDQuery.deserializeBinaryFromReader);
      msg.setGetbysolidityid(value);
      break;
    case 3:
      var value = new ContractCallLocal_pb.ContractCallLocalQuery;
      reader.readMessage(value,ContractCallLocal_pb.ContractCallLocalQuery.deserializeBinaryFromReader);
      msg.setContractcalllocal(value);
      break;
    case 4:
      var value = new ContractGetInfo_pb.ContractGetInfoQuery;
      reader.readMessage(value,ContractGetInfo_pb.ContractGetInfoQuery.deserializeBinaryFromReader);
      msg.setContractgetinfo(value);
      break;
    case 5:
      var value = new ContractGetBytecode_pb.ContractGetBytecodeQuery;
      reader.readMessage(value,ContractGetBytecode_pb.ContractGetBytecodeQuery.deserializeBinaryFromReader);
      msg.setContractgetbytecode(value);
      break;
    case 6:
      var value = new ContractGetRecords_pb.ContractGetRecordsQuery;
      reader.readMessage(value,ContractGetRecords_pb.ContractGetRecordsQuery.deserializeBinaryFromReader);
      msg.setContractgetrecords(value);
      break;
    case 7:
      var value = new CryptoGetAccountBalance_pb.CryptoGetAccountBalanceQuery;
      reader.readMessage(value,CryptoGetAccountBalance_pb.CryptoGetAccountBalanceQuery.deserializeBinaryFromReader);
      msg.setCryptogetaccountbalance(value);
      break;
    case 8:
      var value = new CryptoGetAccountRecords_pb.CryptoGetAccountRecordsQuery;
      reader.readMessage(value,CryptoGetAccountRecords_pb.CryptoGetAccountRecordsQuery.deserializeBinaryFromReader);
      msg.setCryptogetaccountrecords(value);
      break;
    case 9:
      var value = new CryptoGetInfo_pb.CryptoGetInfoQuery;
      reader.readMessage(value,CryptoGetInfo_pb.CryptoGetInfoQuery.deserializeBinaryFromReader);
      msg.setCryptogetinfo(value);
      break;
    case 10:
      var value = new CryptoGetLiveHash_pb.CryptoGetLiveHashQuery;
      reader.readMessage(value,CryptoGetLiveHash_pb.CryptoGetLiveHashQuery.deserializeBinaryFromReader);
      msg.setCryptogetlivehash(value);
      break;
    case 11:
      var value = new CryptoGetStakers_pb.CryptoGetStakersQuery;
      reader.readMessage(value,CryptoGetStakers_pb.CryptoGetStakersQuery.deserializeBinaryFromReader);
      msg.setCryptogetproxystakers(value);
      break;
    case 12:
      var value = new FileGetContents_pb.FileGetContentsQuery;
      reader.readMessage(value,FileGetContents_pb.FileGetContentsQuery.deserializeBinaryFromReader);
      msg.setFilegetcontents(value);
      break;
    case 13:
      var value = new FileGetInfo_pb.FileGetInfoQuery;
      reader.readMessage(value,FileGetInfo_pb.FileGetInfoQuery.deserializeBinaryFromReader);
      msg.setFilegetinfo(value);
      break;
    case 14:
      var value = new TransactionGetReceipt_pb.TransactionGetReceiptQuery;
      reader.readMessage(value,TransactionGetReceipt_pb.TransactionGetReceiptQuery.deserializeBinaryFromReader);
      msg.setTransactiongetreceipt(value);
      break;
    case 15:
      var value = new TransactionGetRecord_pb.TransactionGetRecordQuery;
      reader.readMessage(value,TransactionGetRecord_pb.TransactionGetRecordQuery.deserializeBinaryFromReader);
      msg.setTransactiongetrecord(value);
      break;
    case 16:
      var value = new TransactionGetFastRecord_pb.TransactionGetFastRecordQuery;
      reader.readMessage(value,TransactionGetFastRecord_pb.TransactionGetFastRecordQuery.deserializeBinaryFromReader);
      msg.setTransactiongetfastrecord(value);
      break;
    case 50:
      var value = new ConsensusGetTopicInfo_pb.ConsensusGetTopicInfoQuery;
      reader.readMessage(value,ConsensusGetTopicInfo_pb.ConsensusGetTopicInfoQuery.deserializeBinaryFromReader);
      msg.setConsensusgettopicinfo(value);
      break;
    case 51:
      var value = new NetworkGetVersionInfo_pb.NetworkGetVersionInfoQuery;
      reader.readMessage(value,NetworkGetVersionInfo_pb.NetworkGetVersionInfoQuery.deserializeBinaryFromReader);
      msg.setNetworkgetversioninfo(value);
      break;
    case 52:
      var value = new TokenGetInfo_pb.TokenGetInfoQuery;
      reader.readMessage(value,TokenGetInfo_pb.TokenGetInfoQuery.deserializeBinaryFromReader);
      msg.setTokengetinfo(value);
      break;
    case 53:
      var value = new ScheduleGetInfo_pb.ScheduleGetInfoQuery;
      reader.readMessage(value,ScheduleGetInfo_pb.ScheduleGetInfoQuery.deserializeBinaryFromReader);
      msg.setSchedulegetinfo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.Query.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.Query.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.Query} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.Query.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getGetbykey();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      GetByKey_pb.GetByKeyQuery.serializeBinaryToWriter
    );
  }
  f = message.getGetbysolidityid();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      GetBySolidityID_pb.GetBySolidityIDQuery.serializeBinaryToWriter
    );
  }
  f = message.getContractcalllocal();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      ContractCallLocal_pb.ContractCallLocalQuery.serializeBinaryToWriter
    );
  }
  f = message.getContractgetinfo();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      ContractGetInfo_pb.ContractGetInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getContractgetbytecode();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      ContractGetBytecode_pb.ContractGetBytecodeQuery.serializeBinaryToWriter
    );
  }
  f = message.getContractgetrecords();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      ContractGetRecords_pb.ContractGetRecordsQuery.serializeBinaryToWriter
    );
  }
  f = message.getCryptogetaccountbalance();
  if (f != null) {
    writer.writeMessage(
      7,
      f,
      CryptoGetAccountBalance_pb.CryptoGetAccountBalanceQuery.serializeBinaryToWriter
    );
  }
  f = message.getCryptogetaccountrecords();
  if (f != null) {
    writer.writeMessage(
      8,
      f,
      CryptoGetAccountRecords_pb.CryptoGetAccountRecordsQuery.serializeBinaryToWriter
    );
  }
  f = message.getCryptogetinfo();
  if (f != null) {
    writer.writeMessage(
      9,
      f,
      CryptoGetInfo_pb.CryptoGetInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getCryptogetlivehash();
  if (f != null) {
    writer.writeMessage(
      10,
      f,
      CryptoGetLiveHash_pb.CryptoGetLiveHashQuery.serializeBinaryToWriter
    );
  }
  f = message.getCryptogetproxystakers();
  if (f != null) {
    writer.writeMessage(
      11,
      f,
      CryptoGetStakers_pb.CryptoGetStakersQuery.serializeBinaryToWriter
    );
  }
  f = message.getFilegetcontents();
  if (f != null) {
    writer.writeMessage(
      12,
      f,
      FileGetContents_pb.FileGetContentsQuery.serializeBinaryToWriter
    );
  }
  f = message.getFilegetinfo();
  if (f != null) {
    writer.writeMessage(
      13,
      f,
      FileGetInfo_pb.FileGetInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getTransactiongetreceipt();
  if (f != null) {
    writer.writeMessage(
      14,
      f,
      TransactionGetReceipt_pb.TransactionGetReceiptQuery.serializeBinaryToWriter
    );
  }
  f = message.getTransactiongetrecord();
  if (f != null) {
    writer.writeMessage(
      15,
      f,
      TransactionGetRecord_pb.TransactionGetRecordQuery.serializeBinaryToWriter
    );
  }
  f = message.getTransactiongetfastrecord();
  if (f != null) {
    writer.writeMessage(
      16,
      f,
      TransactionGetFastRecord_pb.TransactionGetFastRecordQuery.serializeBinaryToWriter
    );
  }
  f = message.getConsensusgettopicinfo();
  if (f != null) {
    writer.writeMessage(
      50,
      f,
      ConsensusGetTopicInfo_pb.ConsensusGetTopicInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getNetworkgetversioninfo();
  if (f != null) {
    writer.writeMessage(
      51,
      f,
      NetworkGetVersionInfo_pb.NetworkGetVersionInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getTokengetinfo();
  if (f != null) {
    writer.writeMessage(
      52,
      f,
      TokenGetInfo_pb.TokenGetInfoQuery.serializeBinaryToWriter
    );
  }
  f = message.getSchedulegetinfo();
  if (f != null) {
    writer.writeMessage(
      53,
      f,
      ScheduleGetInfo_pb.ScheduleGetInfoQuery.serializeBinaryToWriter
    );
  }
};


/**
 * optional GetByKeyQuery getByKey = 1;
 * @return {?proto.proto.GetByKeyQuery}
 */
proto.proto.Query.prototype.getGetbykey = function() {
  return /** @type{?proto.proto.GetByKeyQuery} */ (
    jspb.Message.getWrapperField(this, GetByKey_pb.GetByKeyQuery, 1));
};


/**
 * @param {?proto.proto.GetByKeyQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setGetbykey = function(value) {
  return jspb.Message.setOneofWrapperField(this, 1, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearGetbykey = function() {
  return this.setGetbykey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasGetbykey = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional GetBySolidityIDQuery getBySolidityID = 2;
 * @return {?proto.proto.GetBySolidityIDQuery}
 */
proto.proto.Query.prototype.getGetbysolidityid = function() {
  return /** @type{?proto.proto.GetBySolidityIDQuery} */ (
    jspb.Message.getWrapperField(this, GetBySolidityID_pb.GetBySolidityIDQuery, 2));
};


/**
 * @param {?proto.proto.GetBySolidityIDQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setGetbysolidityid = function(value) {
  return jspb.Message.setOneofWrapperField(this, 2, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearGetbysolidityid = function() {
  return this.setGetbysolidityid(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasGetbysolidityid = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional ContractCallLocalQuery contractCallLocal = 3;
 * @return {?proto.proto.ContractCallLocalQuery}
 */
proto.proto.Query.prototype.getContractcalllocal = function() {
  return /** @type{?proto.proto.ContractCallLocalQuery} */ (
    jspb.Message.getWrapperField(this, ContractCallLocal_pb.ContractCallLocalQuery, 3));
};


/**
 * @param {?proto.proto.ContractCallLocalQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setContractcalllocal = function(value) {
  return jspb.Message.setOneofWrapperField(this, 3, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearContractcalllocal = function() {
  return this.setContractcalllocal(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasContractcalllocal = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional ContractGetInfoQuery contractGetInfo = 4;
 * @return {?proto.proto.ContractGetInfoQuery}
 */
proto.proto.Query.prototype.getContractgetinfo = function() {
  return /** @type{?proto.proto.ContractGetInfoQuery} */ (
    jspb.Message.getWrapperField(this, ContractGetInfo_pb.ContractGetInfoQuery, 4));
};


/**
 * @param {?proto.proto.ContractGetInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setContractgetinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 4, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearContractgetinfo = function() {
  return this.setContractgetinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasContractgetinfo = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional ContractGetBytecodeQuery contractGetBytecode = 5;
 * @return {?proto.proto.ContractGetBytecodeQuery}
 */
proto.proto.Query.prototype.getContractgetbytecode = function() {
  return /** @type{?proto.proto.ContractGetBytecodeQuery} */ (
    jspb.Message.getWrapperField(this, ContractGetBytecode_pb.ContractGetBytecodeQuery, 5));
};


/**
 * @param {?proto.proto.ContractGetBytecodeQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setContractgetbytecode = function(value) {
  return jspb.Message.setOneofWrapperField(this, 5, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearContractgetbytecode = function() {
  return this.setContractgetbytecode(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasContractgetbytecode = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional ContractGetRecordsQuery ContractGetRecords = 6;
 * @return {?proto.proto.ContractGetRecordsQuery}
 */
proto.proto.Query.prototype.getContractgetrecords = function() {
  return /** @type{?proto.proto.ContractGetRecordsQuery} */ (
    jspb.Message.getWrapperField(this, ContractGetRecords_pb.ContractGetRecordsQuery, 6));
};


/**
 * @param {?proto.proto.ContractGetRecordsQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setContractgetrecords = function(value) {
  return jspb.Message.setOneofWrapperField(this, 6, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearContractgetrecords = function() {
  return this.setContractgetrecords(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasContractgetrecords = function() {
  return jspb.Message.getField(this, 6) != null;
};


/**
 * optional CryptoGetAccountBalanceQuery cryptogetAccountBalance = 7;
 * @return {?proto.proto.CryptoGetAccountBalanceQuery}
 */
proto.proto.Query.prototype.getCryptogetaccountbalance = function() {
  return /** @type{?proto.proto.CryptoGetAccountBalanceQuery} */ (
    jspb.Message.getWrapperField(this, CryptoGetAccountBalance_pb.CryptoGetAccountBalanceQuery, 7));
};


/**
 * @param {?proto.proto.CryptoGetAccountBalanceQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setCryptogetaccountbalance = function(value) {
  return jspb.Message.setOneofWrapperField(this, 7, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearCryptogetaccountbalance = function() {
  return this.setCryptogetaccountbalance(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasCryptogetaccountbalance = function() {
  return jspb.Message.getField(this, 7) != null;
};


/**
 * optional CryptoGetAccountRecordsQuery cryptoGetAccountRecords = 8;
 * @return {?proto.proto.CryptoGetAccountRecordsQuery}
 */
proto.proto.Query.prototype.getCryptogetaccountrecords = function() {
  return /** @type{?proto.proto.CryptoGetAccountRecordsQuery} */ (
    jspb.Message.getWrapperField(this, CryptoGetAccountRecords_pb.CryptoGetAccountRecordsQuery, 8));
};


/**
 * @param {?proto.proto.CryptoGetAccountRecordsQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setCryptogetaccountrecords = function(value) {
  return jspb.Message.setOneofWrapperField(this, 8, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearCryptogetaccountrecords = function() {
  return this.setCryptogetaccountrecords(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasCryptogetaccountrecords = function() {
  return jspb.Message.getField(this, 8) != null;
};


/**
 * optional CryptoGetInfoQuery cryptoGetInfo = 9;
 * @return {?proto.proto.CryptoGetInfoQuery}
 */
proto.proto.Query.prototype.getCryptogetinfo = function() {
  return /** @type{?proto.proto.CryptoGetInfoQuery} */ (
    jspb.Message.getWrapperField(this, CryptoGetInfo_pb.CryptoGetInfoQuery, 9));
};


/**
 * @param {?proto.proto.CryptoGetInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setCryptogetinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 9, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearCryptogetinfo = function() {
  return this.setCryptogetinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasCryptogetinfo = function() {
  return jspb.Message.getField(this, 9) != null;
};


/**
 * optional CryptoGetLiveHashQuery cryptoGetLiveHash = 10;
 * @return {?proto.proto.CryptoGetLiveHashQuery}
 */
proto.proto.Query.prototype.getCryptogetlivehash = function() {
  return /** @type{?proto.proto.CryptoGetLiveHashQuery} */ (
    jspb.Message.getWrapperField(this, CryptoGetLiveHash_pb.CryptoGetLiveHashQuery, 10));
};


/**
 * @param {?proto.proto.CryptoGetLiveHashQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setCryptogetlivehash = function(value) {
  return jspb.Message.setOneofWrapperField(this, 10, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearCryptogetlivehash = function() {
  return this.setCryptogetlivehash(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasCryptogetlivehash = function() {
  return jspb.Message.getField(this, 10) != null;
};


/**
 * optional CryptoGetStakersQuery cryptoGetProxyStakers = 11;
 * @return {?proto.proto.CryptoGetStakersQuery}
 */
proto.proto.Query.prototype.getCryptogetproxystakers = function() {
  return /** @type{?proto.proto.CryptoGetStakersQuery} */ (
    jspb.Message.getWrapperField(this, CryptoGetStakers_pb.CryptoGetStakersQuery, 11));
};


/**
 * @param {?proto.proto.CryptoGetStakersQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setCryptogetproxystakers = function(value) {
  return jspb.Message.setOneofWrapperField(this, 11, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearCryptogetproxystakers = function() {
  return this.setCryptogetproxystakers(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasCryptogetproxystakers = function() {
  return jspb.Message.getField(this, 11) != null;
};


/**
 * optional FileGetContentsQuery fileGetContents = 12;
 * @return {?proto.proto.FileGetContentsQuery}
 */
proto.proto.Query.prototype.getFilegetcontents = function() {
  return /** @type{?proto.proto.FileGetContentsQuery} */ (
    jspb.Message.getWrapperField(this, FileGetContents_pb.FileGetContentsQuery, 12));
};


/**
 * @param {?proto.proto.FileGetContentsQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setFilegetcontents = function(value) {
  return jspb.Message.setOneofWrapperField(this, 12, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearFilegetcontents = function() {
  return this.setFilegetcontents(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasFilegetcontents = function() {
  return jspb.Message.getField(this, 12) != null;
};


/**
 * optional FileGetInfoQuery fileGetInfo = 13;
 * @return {?proto.proto.FileGetInfoQuery}
 */
proto.proto.Query.prototype.getFilegetinfo = function() {
  return /** @type{?proto.proto.FileGetInfoQuery} */ (
    jspb.Message.getWrapperField(this, FileGetInfo_pb.FileGetInfoQuery, 13));
};


/**
 * @param {?proto.proto.FileGetInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setFilegetinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 13, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearFilegetinfo = function() {
  return this.setFilegetinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasFilegetinfo = function() {
  return jspb.Message.getField(this, 13) != null;
};


/**
 * optional TransactionGetReceiptQuery transactionGetReceipt = 14;
 * @return {?proto.proto.TransactionGetReceiptQuery}
 */
proto.proto.Query.prototype.getTransactiongetreceipt = function() {
  return /** @type{?proto.proto.TransactionGetReceiptQuery} */ (
    jspb.Message.getWrapperField(this, TransactionGetReceipt_pb.TransactionGetReceiptQuery, 14));
};


/**
 * @param {?proto.proto.TransactionGetReceiptQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setTransactiongetreceipt = function(value) {
  return jspb.Message.setOneofWrapperField(this, 14, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearTransactiongetreceipt = function() {
  return this.setTransactiongetreceipt(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasTransactiongetreceipt = function() {
  return jspb.Message.getField(this, 14) != null;
};


/**
 * optional TransactionGetRecordQuery transactionGetRecord = 15;
 * @return {?proto.proto.TransactionGetRecordQuery}
 */
proto.proto.Query.prototype.getTransactiongetrecord = function() {
  return /** @type{?proto.proto.TransactionGetRecordQuery} */ (
    jspb.Message.getWrapperField(this, TransactionGetRecord_pb.TransactionGetRecordQuery, 15));
};


/**
 * @param {?proto.proto.TransactionGetRecordQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setTransactiongetrecord = function(value) {
  return jspb.Message.setOneofWrapperField(this, 15, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearTransactiongetrecord = function() {
  return this.setTransactiongetrecord(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasTransactiongetrecord = function() {
  return jspb.Message.getField(this, 15) != null;
};


/**
 * optional TransactionGetFastRecordQuery transactionGetFastRecord = 16;
 * @return {?proto.proto.TransactionGetFastRecordQuery}
 */
proto.proto.Query.prototype.getTransactiongetfastrecord = function() {
  return /** @type{?proto.proto.TransactionGetFastRecordQuery} */ (
    jspb.Message.getWrapperField(this, TransactionGetFastRecord_pb.TransactionGetFastRecordQuery, 16));
};


/**
 * @param {?proto.proto.TransactionGetFastRecordQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setTransactiongetfastrecord = function(value) {
  return jspb.Message.setOneofWrapperField(this, 16, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearTransactiongetfastrecord = function() {
  return this.setTransactiongetfastrecord(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasTransactiongetfastrecord = function() {
  return jspb.Message.getField(this, 16) != null;
};


/**
 * optional ConsensusGetTopicInfoQuery consensusGetTopicInfo = 50;
 * @return {?proto.proto.ConsensusGetTopicInfoQuery}
 */
proto.proto.Query.prototype.getConsensusgettopicinfo = function() {
  return /** @type{?proto.proto.ConsensusGetTopicInfoQuery} */ (
    jspb.Message.getWrapperField(this, ConsensusGetTopicInfo_pb.ConsensusGetTopicInfoQuery, 50));
};


/**
 * @param {?proto.proto.ConsensusGetTopicInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setConsensusgettopicinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 50, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearConsensusgettopicinfo = function() {
  return this.setConsensusgettopicinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasConsensusgettopicinfo = function() {
  return jspb.Message.getField(this, 50) != null;
};


/**
 * optional NetworkGetVersionInfoQuery networkGetVersionInfo = 51;
 * @return {?proto.proto.NetworkGetVersionInfoQuery}
 */
proto.proto.Query.prototype.getNetworkgetversioninfo = function() {
  return /** @type{?proto.proto.NetworkGetVersionInfoQuery} */ (
    jspb.Message.getWrapperField(this, NetworkGetVersionInfo_pb.NetworkGetVersionInfoQuery, 51));
};


/**
 * @param {?proto.proto.NetworkGetVersionInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setNetworkgetversioninfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 51, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearNetworkgetversioninfo = function() {
  return this.setNetworkgetversioninfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasNetworkgetversioninfo = function() {
  return jspb.Message.getField(this, 51) != null;
};


/**
 * optional TokenGetInfoQuery tokenGetInfo = 52;
 * @return {?proto.proto.TokenGetInfoQuery}
 */
proto.proto.Query.prototype.getTokengetinfo = function() {
  return /** @type{?proto.proto.TokenGetInfoQuery} */ (
    jspb.Message.getWrapperField(this, TokenGetInfo_pb.TokenGetInfoQuery, 52));
};


/**
 * @param {?proto.proto.TokenGetInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setTokengetinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 52, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearTokengetinfo = function() {
  return this.setTokengetinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasTokengetinfo = function() {
  return jspb.Message.getField(this, 52) != null;
};


/**
 * optional ScheduleGetInfoQuery scheduleGetInfo = 53;
 * @return {?proto.proto.ScheduleGetInfoQuery}
 */
proto.proto.Query.prototype.getSchedulegetinfo = function() {
  return /** @type{?proto.proto.ScheduleGetInfoQuery} */ (
    jspb.Message.getWrapperField(this, ScheduleGetInfo_pb.ScheduleGetInfoQuery, 53));
};


/**
 * @param {?proto.proto.ScheduleGetInfoQuery|undefined} value
 * @return {!proto.proto.Query} returns this
*/
proto.proto.Query.prototype.setSchedulegetinfo = function(value) {
  return jspb.Message.setOneofWrapperField(this, 53, proto.proto.Query.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.Query} returns this
 */
proto.proto.Query.prototype.clearSchedulegetinfo = function() {
  return this.setSchedulegetinfo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.Query.prototype.hasSchedulegetinfo = function() {
  return jspb.Message.getField(this, 53) != null;
};


goog.object.extend(exports, proto.proto);
