// @hashgraph/sdk/lib/generated/ScheduleCreate_pb.js
// source: ScheduleCreate.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
var SchedulableTransactionBody_pb = require('./SchedulableTransactionBody_pb.js');
goog.object.extend(proto, SchedulableTransactionBody_pb);
goog.exportSymbol('proto.proto.ScheduleCreateTransactionBody', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ScheduleCreateTransactionBody = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.ScheduleCreateTransactionBody, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ScheduleCreateTransactionBody.displayName = 'proto.proto.ScheduleCreateTransactionBody';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ScheduleCreateTransactionBody.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ScheduleCreateTransactionBody} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ScheduleCreateTransactionBody.toObject = function(includeInstance, msg) {
  var f, obj = {
    scheduledtransactionbody: (f = msg.getScheduledtransactionbody()) && SchedulableTransactionBody_pb.SchedulableTransactionBody.toObject(includeInstance, f),
    memo: jspb.Message.getFieldWithDefault(msg, 2, ""),
    adminkey: (f = msg.getAdminkey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    payeraccountid: (f = msg.getPayeraccountid()) && BasicTypes_pb.AccountID.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ScheduleCreateTransactionBody}
 */
proto.proto.ScheduleCreateTransactionBody.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ScheduleCreateTransactionBody;
  return proto.proto.ScheduleCreateTransactionBody.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ScheduleCreateTransactionBody} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ScheduleCreateTransactionBody}
 */
proto.proto.ScheduleCreateTransactionBody.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new SchedulableTransactionBody_pb.SchedulableTransactionBody;
      reader.readMessage(value,SchedulableTransactionBody_pb.SchedulableTransactionBody.deserializeBinaryFromReader);
      msg.setScheduledtransactionbody(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setMemo(value);
      break;
    case 3:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setAdminkey(value);
      break;
    case 4:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setPayeraccountid(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ScheduleCreateTransactionBody.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ScheduleCreateTransactionBody} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ScheduleCreateTransactionBody.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getScheduledtransactionbody();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      SchedulableTransactionBody_pb.SchedulableTransactionBody.serializeBinaryToWriter
    );
  }
  f = message.getMemo();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getAdminkey();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getPayeraccountid();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
};


/**
 * optional SchedulableTransactionBody scheduledTransactionBody = 1;
 * @return {?proto.proto.SchedulableTransactionBody}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.getScheduledtransactionbody = function() {
  return /** @type{?proto.proto.SchedulableTransactionBody} */ (
    jspb.Message.getWrapperField(this, SchedulableTransactionBody_pb.SchedulableTransactionBody, 1));
};


/**
 * @param {?proto.proto.SchedulableTransactionBody|undefined} value
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
*/
proto.proto.ScheduleCreateTransactionBody.prototype.setScheduledtransactionbody = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
 */
proto.proto.ScheduleCreateTransactionBody.prototype.clearScheduledtransactionbody = function() {
  return this.setScheduledtransactionbody(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.hasScheduledtransactionbody = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string memo = 2;
 * @return {string}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.getMemo = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
 */
proto.proto.ScheduleCreateTransactionBody.prototype.setMemo = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional Key adminKey = 3;
 * @return {?proto.proto.Key}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.getAdminkey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 3));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
*/
proto.proto.ScheduleCreateTransactionBody.prototype.setAdminkey = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
 */
proto.proto.ScheduleCreateTransactionBody.prototype.clearAdminkey = function() {
  return this.setAdminkey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.hasAdminkey = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional AccountID payerAccountID = 4;
 * @return {?proto.proto.AccountID}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.getPayeraccountid = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 4));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
*/
proto.proto.ScheduleCreateTransactionBody.prototype.setPayeraccountid = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.ScheduleCreateTransactionBody} returns this
 */
proto.proto.ScheduleCreateTransactionBody.prototype.clearPayeraccountid = function() {
  return this.setPayeraccountid(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.ScheduleCreateTransactionBody.prototype.hasPayeraccountid = function() {
  return jspb.Message.getField(this, 4) != null;
};


goog.object.extend(exports, proto.proto);
