// @hashgraph/sdk/lib/generated/ThrottleDefinitions_pb.js
// source: ThrottleDefinitions.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
goog.exportSymbol('proto.proto.ThrottleBucket', null, global);
goog.exportSymbol('proto.proto.ThrottleDefinitions', null, global);
goog.exportSymbol('proto.proto.ThrottleGroup', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ThrottleGroup = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.ThrottleGroup.repeatedFields_, null);
};
goog.inherits(proto.proto.ThrottleGroup, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ThrottleGroup.displayName = 'proto.proto.ThrottleGroup';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ThrottleBucket = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.ThrottleBucket.repeatedFields_, null);
};
goog.inherits(proto.proto.ThrottleBucket, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ThrottleBucket.displayName = 'proto.proto.ThrottleBucket';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.ThrottleDefinitions = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.ThrottleDefinitions.repeatedFields_, null);
};
goog.inherits(proto.proto.ThrottleDefinitions, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.ThrottleDefinitions.displayName = 'proto.proto.ThrottleDefinitions';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.ThrottleGroup.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ThrottleGroup.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ThrottleGroup.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ThrottleGroup} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleGroup.toObject = function(includeInstance, msg) {
  var f, obj = {
    operationsList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    milliopspersec: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ThrottleGroup}
 */
proto.proto.ThrottleGroup.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ThrottleGroup;
  return proto.proto.ThrottleGroup.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ThrottleGroup} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ThrottleGroup}
 */
proto.proto.ThrottleGroup.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Array<!proto.proto.HederaFunctionality>} */ (reader.readPackedEnum());
      msg.setOperationsList(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setMilliopspersec(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ThrottleGroup.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ThrottleGroup.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ThrottleGroup} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleGroup.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getOperationsList();
  if (f.length > 0) {
    writer.writePackedEnum(
      1,
      f
    );
  }
  f = message.getMilliopspersec();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
};


/**
 * repeated HederaFunctionality operations = 1;
 * @return {!Array<!proto.proto.HederaFunctionality>}
 */
proto.proto.ThrottleGroup.prototype.getOperationsList = function() {
  return /** @type {!Array<!proto.proto.HederaFunctionality>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<!proto.proto.HederaFunctionality>} value
 * @return {!proto.proto.ThrottleGroup} returns this
 */
proto.proto.ThrottleGroup.prototype.setOperationsList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {!proto.proto.HederaFunctionality} value
 * @param {number=} opt_index
 * @return {!proto.proto.ThrottleGroup} returns this
 */
proto.proto.ThrottleGroup.prototype.addOperations = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.ThrottleGroup} returns this
 */
proto.proto.ThrottleGroup.prototype.clearOperationsList = function() {
  return this.setOperationsList([]);
};


/**
 * optional uint64 milliOpsPerSec = 2;
 * @return {number}
 */
proto.proto.ThrottleGroup.prototype.getMilliopspersec = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.proto.ThrottleGroup} returns this
 */
proto.proto.ThrottleGroup.prototype.setMilliopspersec = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.ThrottleBucket.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ThrottleBucket.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ThrottleBucket.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ThrottleBucket} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleBucket.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    burstperiodms: jspb.Message.getFieldWithDefault(msg, 2, 0),
    throttlegroupsList: jspb.Message.toObjectList(msg.getThrottlegroupsList(),
    proto.proto.ThrottleGroup.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ThrottleBucket}
 */
proto.proto.ThrottleBucket.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ThrottleBucket;
  return proto.proto.ThrottleBucket.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ThrottleBucket} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ThrottleBucket}
 */
proto.proto.ThrottleBucket.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint64());
      msg.setBurstperiodms(value);
      break;
    case 3:
      var value = new proto.proto.ThrottleGroup;
      reader.readMessage(value,proto.proto.ThrottleGroup.deserializeBinaryFromReader);
      msg.addThrottlegroups(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ThrottleBucket.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ThrottleBucket.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ThrottleBucket} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleBucket.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getBurstperiodms();
  if (f !== 0) {
    writer.writeUint64(
      2,
      f
    );
  }
  f = message.getThrottlegroupsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      proto.proto.ThrottleGroup.serializeBinaryToWriter
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.proto.ThrottleBucket.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.ThrottleBucket} returns this
 */
proto.proto.ThrottleBucket.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint64 burstPeriodMs = 2;
 * @return {number}
 */
proto.proto.ThrottleBucket.prototype.getBurstperiodms = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.proto.ThrottleBucket} returns this
 */
proto.proto.ThrottleBucket.prototype.setBurstperiodms = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * repeated ThrottleGroup throttleGroups = 3;
 * @return {!Array<!proto.proto.ThrottleGroup>}
 */
proto.proto.ThrottleBucket.prototype.getThrottlegroupsList = function() {
  return /** @type{!Array<!proto.proto.ThrottleGroup>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.proto.ThrottleGroup, 3));
};


/**
 * @param {!Array<!proto.proto.ThrottleGroup>} value
 * @return {!proto.proto.ThrottleBucket} returns this
*/
proto.proto.ThrottleBucket.prototype.setThrottlegroupsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.proto.ThrottleGroup=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.ThrottleGroup}
 */
proto.proto.ThrottleBucket.prototype.addThrottlegroups = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.proto.ThrottleGroup, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.ThrottleBucket} returns this
 */
proto.proto.ThrottleBucket.prototype.clearThrottlegroupsList = function() {
  return this.setThrottlegroupsList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.ThrottleDefinitions.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.ThrottleDefinitions.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.ThrottleDefinitions.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.ThrottleDefinitions} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleDefinitions.toObject = function(includeInstance, msg) {
  var f, obj = {
    throttlebucketsList: jspb.Message.toObjectList(msg.getThrottlebucketsList(),
    proto.proto.ThrottleBucket.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.ThrottleDefinitions}
 */
proto.proto.ThrottleDefinitions.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.ThrottleDefinitions;
  return proto.proto.ThrottleDefinitions.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.ThrottleDefinitions} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.ThrottleDefinitions}
 */
proto.proto.ThrottleDefinitions.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.proto.ThrottleBucket;
      reader.readMessage(value,proto.proto.ThrottleBucket.deserializeBinaryFromReader);
      msg.addThrottlebuckets(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.ThrottleDefinitions.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.ThrottleDefinitions.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.ThrottleDefinitions} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.ThrottleDefinitions.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getThrottlebucketsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.proto.ThrottleBucket.serializeBinaryToWriter
    );
  }
};


/**
 * repeated ThrottleBucket throttleBuckets = 1;
 * @return {!Array<!proto.proto.ThrottleBucket>}
 */
proto.proto.ThrottleDefinitions.prototype.getThrottlebucketsList = function() {
  return /** @type{!Array<!proto.proto.ThrottleBucket>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.proto.ThrottleBucket, 1));
};


/**
 * @param {!Array<!proto.proto.ThrottleBucket>} value
 * @return {!proto.proto.ThrottleDefinitions} returns this
*/
proto.proto.ThrottleDefinitions.prototype.setThrottlebucketsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.proto.ThrottleBucket=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.ThrottleBucket}
 */
proto.proto.ThrottleDefinitions.prototype.addThrottlebuckets = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.proto.ThrottleBucket, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.ThrottleDefinitions} returns this
 */
proto.proto.ThrottleDefinitions.prototype.clearThrottlebucketsList = function() {
  return this.setThrottlebucketsList([]);
};


goog.object.extend(exports, proto.proto);
