// @hashgraph/sdk/lib/generated/TokenUpdate_pb.js
// source: TokenUpdate.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
var Duration_pb = require('./Duration_pb.js');
goog.object.extend(proto, Duration_pb);
var Timestamp_pb = require('./Timestamp_pb.js');
goog.object.extend(proto, Timestamp_pb);
var google_protobuf_wrappers_pb = require('google-protobuf/google/protobuf/wrappers_pb.js');
goog.object.extend(proto, google_protobuf_wrappers_pb);
goog.exportSymbol('proto.proto.TokenUpdateTransactionBody', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.TokenUpdateTransactionBody = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.TokenUpdateTransactionBody, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.TokenUpdateTransactionBody.displayName = 'proto.proto.TokenUpdateTransactionBody';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.TokenUpdateTransactionBody.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.TokenUpdateTransactionBody.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.TokenUpdateTransactionBody} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.TokenUpdateTransactionBody.toObject = function(includeInstance, msg) {
  var f, obj = {
    token: (f = msg.getToken()) && BasicTypes_pb.TokenID.toObject(includeInstance, f),
    symbol: jspb.Message.getFieldWithDefault(msg, 2, ""),
    name: jspb.Message.getFieldWithDefault(msg, 3, ""),
    treasury: (f = msg.getTreasury()) && BasicTypes_pb.AccountID.toObject(includeInstance, f),
    adminkey: (f = msg.getAdminkey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    kyckey: (f = msg.getKyckey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    freezekey: (f = msg.getFreezekey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    wipekey: (f = msg.getWipekey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    supplykey: (f = msg.getSupplykey()) && BasicTypes_pb.Key.toObject(includeInstance, f),
    autorenewaccount: (f = msg.getAutorenewaccount()) && BasicTypes_pb.AccountID.toObject(includeInstance, f),
    autorenewperiod: (f = msg.getAutorenewperiod()) && Duration_pb.Duration.toObject(includeInstance, f),
    expiry: (f = msg.getExpiry()) && Timestamp_pb.Timestamp.toObject(includeInstance, f),
    memo: (f = msg.getMemo()) && google_protobuf_wrappers_pb.StringValue.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.TokenUpdateTransactionBody}
 */
proto.proto.TokenUpdateTransactionBody.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.TokenUpdateTransactionBody;
  return proto.proto.TokenUpdateTransactionBody.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.TokenUpdateTransactionBody} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.TokenUpdateTransactionBody}
 */
proto.proto.TokenUpdateTransactionBody.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new BasicTypes_pb.TokenID;
      reader.readMessage(value,BasicTypes_pb.TokenID.deserializeBinaryFromReader);
      msg.setToken(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSymbol(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 4:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setTreasury(value);
      break;
    case 5:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setAdminkey(value);
      break;
    case 6:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setKyckey(value);
      break;
    case 7:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setFreezekey(value);
      break;
    case 8:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setWipekey(value);
      break;
    case 9:
      var value = new BasicTypes_pb.Key;
      reader.readMessage(value,BasicTypes_pb.Key.deserializeBinaryFromReader);
      msg.setSupplykey(value);
      break;
    case 10:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setAutorenewaccount(value);
      break;
    case 11:
      var value = new Duration_pb.Duration;
      reader.readMessage(value,Duration_pb.Duration.deserializeBinaryFromReader);
      msg.setAutorenewperiod(value);
      break;
    case 12:
      var value = new Timestamp_pb.Timestamp;
      reader.readMessage(value,Timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setExpiry(value);
      break;
    case 13:
      var value = new google_protobuf_wrappers_pb.StringValue;
      reader.readMessage(value,google_protobuf_wrappers_pb.StringValue.deserializeBinaryFromReader);
      msg.setMemo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.TokenUpdateTransactionBody.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.TokenUpdateTransactionBody.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.TokenUpdateTransactionBody} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.TokenUpdateTransactionBody.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getToken();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      BasicTypes_pb.TokenID.serializeBinaryToWriter
    );
  }
  f = message.getSymbol();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getTreasury();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
  f = message.getAdminkey();
  if (f != null) {
    writer.writeMessage(
      5,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getKyckey();
  if (f != null) {
    writer.writeMessage(
      6,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getFreezekey();
  if (f != null) {
    writer.writeMessage(
      7,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getWipekey();
  if (f != null) {
    writer.writeMessage(
      8,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getSupplykey();
  if (f != null) {
    writer.writeMessage(
      9,
      f,
      BasicTypes_pb.Key.serializeBinaryToWriter
    );
  }
  f = message.getAutorenewaccount();
  if (f != null) {
    writer.writeMessage(
      10,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
  f = message.getAutorenewperiod();
  if (f != null) {
    writer.writeMessage(
      11,
      f,
      Duration_pb.Duration.serializeBinaryToWriter
    );
  }
  f = message.getExpiry();
  if (f != null) {
    writer.writeMessage(
      12,
      f,
      Timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getMemo();
  if (f != null) {
    writer.writeMessage(
      13,
      f,
      google_protobuf_wrappers_pb.StringValue.serializeBinaryToWriter
    );
  }
};


/**
 * optional TokenID token = 1;
 * @return {?proto.proto.TokenID}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getToken = function() {
  return /** @type{?proto.proto.TokenID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.TokenID, 1));
};


/**
 * @param {?proto.proto.TokenID|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setToken = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearToken = function() {
  return this.setToken(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasToken = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string symbol = 2;
 * @return {string}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getSymbol = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.setSymbol = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string name = 3;
 * @return {string}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional AccountID treasury = 4;
 * @return {?proto.proto.AccountID}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getTreasury = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 4));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setTreasury = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearTreasury = function() {
  return this.setTreasury(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasTreasury = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional Key adminKey = 5;
 * @return {?proto.proto.Key}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getAdminkey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 5));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setAdminkey = function(value) {
  return jspb.Message.setWrapperField(this, 5, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearAdminkey = function() {
  return this.setAdminkey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasAdminkey = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional Key kycKey = 6;
 * @return {?proto.proto.Key}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getKyckey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 6));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setKyckey = function(value) {
  return jspb.Message.setWrapperField(this, 6, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearKyckey = function() {
  return this.setKyckey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasKyckey = function() {
  return jspb.Message.getField(this, 6) != null;
};


/**
 * optional Key freezeKey = 7;
 * @return {?proto.proto.Key}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getFreezekey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 7));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setFreezekey = function(value) {
  return jspb.Message.setWrapperField(this, 7, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearFreezekey = function() {
  return this.setFreezekey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasFreezekey = function() {
  return jspb.Message.getField(this, 7) != null;
};


/**
 * optional Key wipeKey = 8;
 * @return {?proto.proto.Key}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getWipekey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 8));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setWipekey = function(value) {
  return jspb.Message.setWrapperField(this, 8, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearWipekey = function() {
  return this.setWipekey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasWipekey = function() {
  return jspb.Message.getField(this, 8) != null;
};


/**
 * optional Key supplyKey = 9;
 * @return {?proto.proto.Key}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getSupplykey = function() {
  return /** @type{?proto.proto.Key} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.Key, 9));
};


/**
 * @param {?proto.proto.Key|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setSupplykey = function(value) {
  return jspb.Message.setWrapperField(this, 9, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearSupplykey = function() {
  return this.setSupplykey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasSupplykey = function() {
  return jspb.Message.getField(this, 9) != null;
};


/**
 * optional AccountID autoRenewAccount = 10;
 * @return {?proto.proto.AccountID}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getAutorenewaccount = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 10));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setAutorenewaccount = function(value) {
  return jspb.Message.setWrapperField(this, 10, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearAutorenewaccount = function() {
  return this.setAutorenewaccount(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasAutorenewaccount = function() {
  return jspb.Message.getField(this, 10) != null;
};


/**
 * optional Duration autoRenewPeriod = 11;
 * @return {?proto.proto.Duration}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getAutorenewperiod = function() {
  return /** @type{?proto.proto.Duration} */ (
    jspb.Message.getWrapperField(this, Duration_pb.Duration, 11));
};


/**
 * @param {?proto.proto.Duration|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setAutorenewperiod = function(value) {
  return jspb.Message.setWrapperField(this, 11, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearAutorenewperiod = function() {
  return this.setAutorenewperiod(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasAutorenewperiod = function() {
  return jspb.Message.getField(this, 11) != null;
};


/**
 * optional Timestamp expiry = 12;
 * @return {?proto.proto.Timestamp}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getExpiry = function() {
  return /** @type{?proto.proto.Timestamp} */ (
    jspb.Message.getWrapperField(this, Timestamp_pb.Timestamp, 12));
};


/**
 * @param {?proto.proto.Timestamp|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setExpiry = function(value) {
  return jspb.Message.setWrapperField(this, 12, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearExpiry = function() {
  return this.setExpiry(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasExpiry = function() {
  return jspb.Message.getField(this, 12) != null;
};


/**
 * optional google.protobuf.StringValue memo = 13;
 * @return {?proto.google.protobuf.StringValue}
 */
proto.proto.TokenUpdateTransactionBody.prototype.getMemo = function() {
  return /** @type{?proto.google.protobuf.StringValue} */ (
    jspb.Message.getWrapperField(this, google_protobuf_wrappers_pb.StringValue, 13));
};


/**
 * @param {?proto.google.protobuf.StringValue|undefined} value
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
*/
proto.proto.TokenUpdateTransactionBody.prototype.setMemo = function(value) {
  return jspb.Message.setWrapperField(this, 13, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenUpdateTransactionBody} returns this
 */
proto.proto.TokenUpdateTransactionBody.prototype.clearMemo = function() {
  return this.setMemo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenUpdateTransactionBody.prototype.hasMemo = function() {
  return jspb.Message.getField(this, 13) != null;
};


goog.object.extend(exports, proto.proto);
