// @hashgraph/sdk/lib/generated/TokenAssociate_pb.js
// source: TokenAssociate.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
goog.exportSymbol('proto.proto.TokenAssociateTransactionBody', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.TokenAssociateTransactionBody = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.proto.TokenAssociateTransactionBody.repeatedFields_, null);
};
goog.inherits(proto.proto.TokenAssociateTransactionBody, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.TokenAssociateTransactionBody.displayName = 'proto.proto.TokenAssociateTransactionBody';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.proto.TokenAssociateTransactionBody.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.TokenAssociateTransactionBody.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.TokenAssociateTransactionBody.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.TokenAssociateTransactionBody} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.TokenAssociateTransactionBody.toObject = function(includeInstance, msg) {
  var f, obj = {
    account: (f = msg.getAccount()) && BasicTypes_pb.AccountID.toObject(includeInstance, f),
    tokensList: jspb.Message.toObjectList(msg.getTokensList(),
    BasicTypes_pb.TokenID.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.TokenAssociateTransactionBody}
 */
proto.proto.TokenAssociateTransactionBody.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.TokenAssociateTransactionBody;
  return proto.proto.TokenAssociateTransactionBody.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.TokenAssociateTransactionBody} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.TokenAssociateTransactionBody}
 */
proto.proto.TokenAssociateTransactionBody.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new BasicTypes_pb.AccountID;
      reader.readMessage(value,BasicTypes_pb.AccountID.deserializeBinaryFromReader);
      msg.setAccount(value);
      break;
    case 2:
      var value = new BasicTypes_pb.TokenID;
      reader.readMessage(value,BasicTypes_pb.TokenID.deserializeBinaryFromReader);
      msg.addTokens(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.TokenAssociateTransactionBody.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.TokenAssociateTransactionBody.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.TokenAssociateTransactionBody} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.TokenAssociateTransactionBody.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAccount();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      BasicTypes_pb.AccountID.serializeBinaryToWriter
    );
  }
  f = message.getTokensList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      BasicTypes_pb.TokenID.serializeBinaryToWriter
    );
  }
};


/**
 * optional AccountID account = 1;
 * @return {?proto.proto.AccountID}
 */
proto.proto.TokenAssociateTransactionBody.prototype.getAccount = function() {
  return /** @type{?proto.proto.AccountID} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.AccountID, 1));
};


/**
 * @param {?proto.proto.AccountID|undefined} value
 * @return {!proto.proto.TokenAssociateTransactionBody} returns this
*/
proto.proto.TokenAssociateTransactionBody.prototype.setAccount = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.TokenAssociateTransactionBody} returns this
 */
proto.proto.TokenAssociateTransactionBody.prototype.clearAccount = function() {
  return this.setAccount(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.TokenAssociateTransactionBody.prototype.hasAccount = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated TokenID tokens = 2;
 * @return {!Array<!proto.proto.TokenID>}
 */
proto.proto.TokenAssociateTransactionBody.prototype.getTokensList = function() {
  return /** @type{!Array<!proto.proto.TokenID>} */ (
    jspb.Message.getRepeatedWrapperField(this, BasicTypes_pb.TokenID, 2));
};


/**
 * @param {!Array<!proto.proto.TokenID>} value
 * @return {!proto.proto.TokenAssociateTransactionBody} returns this
*/
proto.proto.TokenAssociateTransactionBody.prototype.setTokensList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.proto.TokenID=} opt_value
 * @param {number=} opt_index
 * @return {!proto.proto.TokenID}
 */
proto.proto.TokenAssociateTransactionBody.prototype.addTokens = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.proto.TokenID, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.proto.TokenAssociateTransactionBody} returns this
 */
proto.proto.TokenAssociateTransactionBody.prototype.clearTokensList = function() {
  return this.setTokensList([]);
};


goog.object.extend(exports, proto.proto);
