// @hashgraph/sdk/lib/generated/TransactionContents_pb.js
// source: TransactionContents.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var BasicTypes_pb = require('./BasicTypes_pb.js');
goog.object.extend(proto, BasicTypes_pb);
goog.exportSymbol('proto.proto.SignedTransaction', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.proto.SignedTransaction = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.proto.SignedTransaction, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.proto.SignedTransaction.displayName = 'proto.proto.SignedTransaction';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.proto.SignedTransaction.prototype.toObject = function(opt_includeInstance) {
  return proto.proto.SignedTransaction.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.proto.SignedTransaction} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.SignedTransaction.toObject = function(includeInstance, msg) {
  var f, obj = {
    bodybytes: msg.getBodybytes_asB64(),
    sigmap: (f = msg.getSigmap()) && BasicTypes_pb.SignatureMap.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.proto.SignedTransaction}
 */
proto.proto.SignedTransaction.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.proto.SignedTransaction;
  return proto.proto.SignedTransaction.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.proto.SignedTransaction} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.proto.SignedTransaction}
 */
proto.proto.SignedTransaction.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setBodybytes(value);
      break;
    case 2:
      var value = new BasicTypes_pb.SignatureMap;
      reader.readMessage(value,BasicTypes_pb.SignatureMap.deserializeBinaryFromReader);
      msg.setSigmap(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.proto.SignedTransaction.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.proto.SignedTransaction.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.proto.SignedTransaction} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.proto.SignedTransaction.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBodybytes_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      1,
      f
    );
  }
  f = message.getSigmap();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      BasicTypes_pb.SignatureMap.serializeBinaryToWriter
    );
  }
};


/**
 * optional bytes bodyBytes = 1;
 * @return {!(string|Uint8Array)}
 */
proto.proto.SignedTransaction.prototype.getBodybytes = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * optional bytes bodyBytes = 1;
 * This is a type-conversion wrapper around `getBodybytes()`
 * @return {string}
 */
proto.proto.SignedTransaction.prototype.getBodybytes_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getBodybytes()));
};


/**
 * optional bytes bodyBytes = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getBodybytes()`
 * @return {!Uint8Array}
 */
proto.proto.SignedTransaction.prototype.getBodybytes_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getBodybytes()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.proto.SignedTransaction} returns this
 */
proto.proto.SignedTransaction.prototype.setBodybytes = function(value) {
  return jspb.Message.setProto3BytesField(this, 1, value);
};


/**
 * optional SignatureMap sigMap = 2;
 * @return {?proto.proto.SignatureMap}
 */
proto.proto.SignedTransaction.prototype.getSigmap = function() {
  return /** @type{?proto.proto.SignatureMap} */ (
    jspb.Message.getWrapperField(this, BasicTypes_pb.SignatureMap, 2));
};


/**
 * @param {?proto.proto.SignatureMap|undefined} value
 * @return {!proto.proto.SignedTransaction} returns this
*/
proto.proto.SignedTransaction.prototype.setSigmap = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.proto.SignedTransaction} returns this
 */
proto.proto.SignedTransaction.prototype.clearSigmap = function() {
  return this.setSigmap(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.proto.SignedTransaction.prototype.hasSigmap = function() {
  return jspb.Message.getField(this, 2) != null;
};


goog.object.extend(exports, proto.proto);
