// @hashgraph/sdk/lib/token/TokenId.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenId = void 0;
const BasicTypes_pb_1 = require("../generated/BasicTypes_pb");
const util_1 = require("../util");
const bignumber_js_1 = require("bignumber.js");
const hex = require("@stablelib/hex");
/** Normalized token ID returned by various methods in the SDK. */
class TokenId {
    constructor(shardOrTokenId, realm, token) {
        if (typeof shardOrTokenId === "number" &&
            realm != null &&
            token != null) {
            this.shard = shardOrTokenId;
            this.realm = realm;
            this.token = token;
        }
        else {
            const tokenId = shardOrTokenId;
            const id = tokenId instanceof TokenId ?
                tokenId :
                util_1.normalizeEntityId("token", tokenId);
            this.shard = id.shard;
            this.realm = id.realm;
            this.token = id.token;
        }
    }
    static fromString(id) {
        return new TokenId(id);
    }
    // NOT A STABLE API
    static _fromProto(tokenId) {
        return new TokenId({
            shard: tokenId.getShardnum(),
            realm: tokenId.getRealmnum(),
            token: tokenId.getTokennum()
        });
    }
    toString() {
        return `${this.shard}.${this.realm}.${this.token}`;
    }
    static fromSolidityAddress(address) {
        if (address.length !== 40) {
            throw new Error(`Invalid hex encoded solidity address length:
                    expected length 40, got length ${address.length}`);
        }
        // First 4 bytes encoded as 8 characters
        const shard = new bignumber_js_1.default(address.slice(0, 8), 16).toNumber();
        // Next 8 bytes encoded as 16 characters
        const realm = new bignumber_js_1.default(address.slice(8, 24), 16).toNumber();
        // Next 8 bytes encoded as 16 characters
        const token = new bignumber_js_1.default(address.slice(24, 40), 16).toNumber();
        return new TokenId(shard, realm, token);
    }
    toSolidityAddress() {
        const buffer = new Uint8Array(20);
        const view = new DataView(buffer.buffer, 0, 20);
        view.setUint32(0, this.shard);
        view.setUint32(8, this.realm);
        view.setUint32(16, this.token);
        return hex.encode(buffer, true);
    }
    // NOT A STABLE API
    _toProto() {
        const acctId = new BasicTypes_pb_1.TokenID();
        acctId.setShardnum(this.shard);
        acctId.setRealmnum(this.realm);
        acctId.setTokennum(this.token);
        return acctId;
    }
}
exports.TokenId = TokenId;
